worker/src/queues/ingestionFlushQueueExecutor.ts

```diff
@@ -22,93 +18,73 @@ import { SpanKind } from "@opentelemetry/api";
 
 const ingestionFlushQueue = getIngestionFlushQueue();
 
-const createIngestionQueueExecutor = () => {
-  const redisInstance = createNewRedisInstance();
-  if (redisInstance) {
-    return new Worker(
-      QueueName.IngestionFlushQueue,
-      async (job) => {
-        return instrumentAsync(
-          {
-            name: "flush-ingestion-consumer",
-            spanKind: SpanKind.CONSUMER,
-          },
-          async () => {
-            if (job.name === QueueJobs.FlushIngestionEntity) {
-              const flushKey = job.id;
-              if (!flushKey) {
-                throw new Error("Flushkey not provided");
-              }
+export const ingestionFlushQueueProcessor: Processor = async (job) => {
+  return instrumentAsync(
+    {
+      name: "flush-ingestion-consumer",
+      spanKind: SpanKind.CONSUMER,
+    },
+    async () => {
+      if (job.name === QueueJobs.FlushIngestionEntity) {
+        const flushKey = job.id;
+        if (!flushKey) {
+          throw new Error("Flushkey not provided");
+        }
 
-              // Log wait time
-              const waitTime = Date.now() - job.timestamp;
-              logger.debug(
-                `Received flush request after ${waitTime} ms for ${flushKey}`
-              );
+        // Log wait time
+        const waitTime = Date.now() - job.timestamp;
+        logger.debug(
+          `Received flush request after ${waitTime} ms for ${flushKey}`,
+        );
 
-              recordIncrement("ingestion_processing_request");
-              recordHistogram("ingestion_flush_wait_time", waitTime, {
-                unit: "milliseconds",
-              });
+        recordIncrement("ingestion_processing_request");
+        recordHistogram("ingestion_flush_wait_time", waitTime, {
+          unit: "milliseconds",
+        });
 
-              try {
-                // Check dependencies
-                if (!redisInstance) throw new Error("Redis not available");
-                if (!prisma) throw new Error("Prisma not available");
-                if (!ingestionFlushQueue)
-                  throw new Error("Ingestion flush queue not available");
+        try {
+          // Check dependencies
+          if (!redis) throw new Error("Redis not available");
```

Comment:
This is moving from the "shadow" redis instance of the worker to our shared, global redis instance. IMO decoupling the processor logic from the worker itself makes sense. 

An alternative might be that we initialize a new redis connection for each execution. Happy to get some feedback on this.

Reply:
Discussed offline. Change accepted as is.